airbyte-integrations/connectors/source-google-sheets/google_sheets_source/google_sheets_source.py

```diff
@@ -107,9 +107,13 @@ def read(
             while True:
                 range = f"{sheet}!{row_cursor}:{row_cursor + ROW_BATCH_SIZE}"
                 logger.info(f"Fetching range {range}")
-                row_batch = SpreadsheetValues.parse_obj(
-                    client.get_values(spreadsheetId=spreadsheet_id, ranges=range, majorDimension="ROWS")
-                )
+                try:
```

Comment:
can we instead cap the range based on how large the spreadsheet is? e.g: 

```
if row_cursor >= num_rows_in_sheet: 
    break

range = f"{sheet}!{row_cursor}:{min(num_rows_in_sheet, row_cursor+ROW_BATCH_SIZE)}
```

Reply:
Okay, I`ll try make this